lib/engine/game/g_18_oe/step/buy_sell_par_shares.rb

```diff
@@ -16,23 +26,69 @@ def actions(entity)
             actions << 'par' if can_ipo_any?(entity) || can_float_minor?(entity)
             actions << 'buy_company' if !purchasable_companies(entity).empty? || !buyable_bank_owned_companies(entity).empty?
             actions << 'sell_shares' if can_sell_any?(entity)
-
-            actions << 'pass' if !can_float_minor?(entity) && !bought? && !actions.empty?
+            actions << 'convert' unless @can_convert.empty?
+            actions << 'pass' if !can_float_minor?(entity) && !actions.empty?
             actions
           end
 
+          def corporation_actions(entity)
+            return [] unless @can_convert.include?(entity)
+
+            %w[convert pass]
+          end
+
           def can_buy_any_from_ipo?(entity)
             return unless @game.corporations.all?(&:ipoed)
 
             super
           end
 
+          def can_buy?(entity, bundle)
+            action = @round.current_actions.find { |a| a.is_a?(Engine::Action::Convert) }
+            return false if action && action&.entity != bundle.corporation
+
+            super
+          end
+
+          def can_sell?(entity, bundle)
+            return unless bundle
+            return if bundle.corporation.type == :regional
+
+            super
```

Comment:
There needs to be a check here to stop the player selling shares in a major that has just formed.